elfpy/data/contract_interface.py

```diff
@@ -56,17 +67,64 @@ class HyperdriveAddressesJson:
     mock_hyperdrive_math: str = attr.ib()
 
 
-def get_account_balance_for_contract(funding_contract: Contract, account_address: str) -> int:
-    """Return the balance of the account"""
-    return funding_contract.functions.balanceOf(account_address).call()
+def initialize_web3_with_http_provider(ethereum_node: URI | str, request_kwargs: dict | None = None) -> Web3:
+    """Initialize a Web3 instance using an HTTP provider and inject a geth Proof of Authority (poa) middleware.
+
+    Arguments
+    ---------
+    ethereum_node: URI | str
+        Address of the http provider
+    request_kwargs: dict
+        The HTTPProvider uses the python requests library for making requests.
+        If you would like to modify how requests are made,
+        you can use the request_kwargs to do so.
+    """
+    if request_kwargs is None:
+        request_kwargs = {}
```

Comment:
nit, why not default `request_kwargs` to an empty dictionary in the function parameters instead of `None`?

Reply:
This was actually discussed in the discord today. I'll @ you on it, feel free to un-resolve this if you want it to change.